src/jaegertracing/baggage/BaggageTest.cpp

```diff
@@ -21,12 +21,61 @@
 #include "jaegertracing/baggage/RestrictionManager.h"
 #include "jaegertracing/baggage/RestrictionsConfig.h"
 #include "jaegertracing/metrics/Metrics.h"
+#include "jaegertracing/metrics/InMemoryStatsReporter.h"
 #include "jaegertracing/testutils/MockAgent.h"
 
+#include <mutex>
+
 namespace jaegertracing {
 namespace baggage {
 namespace {
 
+class TestStatsReporter : public metrics::StatsReporter {
+  public:
+
+      using ValueMap = std::unordered_map<std::string, int64_t>;
+
+    virtual ~TestStatsReporter() = default;
+
+    void incCounter(const std::string& name,
+                    int64_t delta,
+                    const metrics::StatsReporter::TagMap& tags) override
+    {
+        std::lock_guard<std::mutex> lock(_mutex);
+
+        const auto metricName =
+            metrics::Metrics::addTagsToMetricName(name, tags);
+        auto& currentValue = _counters[metricName];
+        currentValue = currentValue + delta;
+    }
+
+    void recordTimer(const std::string& name,
+                     int64_t time,
+                     const metrics::StatsReporter::TagMap& tags) override
+    {
+    }
+
+    void updateGauge(const std::string& name,
+                     int64_t time,
+                     const metrics::StatsReporter::TagMap& tags) override
+    {
+    }
+
+    int64_t counterValue(const std::string& name,
+                         const metrics::StatsReporter::TagMap& tags)
+    {
+        std::lock_guard<std::mutex> lock(_mutex);
+
+        const auto metricName =
+            metrics::Metrics::addTagsToMetricName(name, tags);
+        return _counters[metricName];
+    }
+
+  private:
+    ValueMap _counters;
+    std::mutex _mutex;
```

Comment:
Use `volatile` keyword to allow for usage in `const` methods.

Reply:
mutable used.